package-manifest.toml

```diff
@@ -134,10 +134,34 @@ commit = "2be097ddd1d3fd8e7f56bc0a4bfd696253b11454"
 # https://buildomat.eng.oxide.computer/public/file/oxidecomputer/maghemite/image/<commit>/mg-ddm.sha256.txt
 sha256 = "94218915ec6fed75dcc81d736bd5e9ef62c9eb651a67a943b297d94d6b390941"
 
-[external_package.dendrite]
+[external_package.dendrite_tofino_stub]
 service_name = "dendrite"
 zone = true
-[external_package.dendrite.source]
+[external_package.dendrite_tofino_stub.only_for_targets]
+switch_variant = "tofino_stub"
+[external_package.dendrite_tofino_stub.source]
+# To manually override the package source:
+#
+# 1. Build the zone image manually
+#   1a. cd <dendrite tree>
+#   1b. cargo build --features=<tofino_stub|tofino_asic|softnpu>
+#   1c. cargo xtask dist -o
+# 2. Copy the output zone image from dendrite/out to omicron/out
+# 3. Use type = "manual" instead of the "prebuilt"
+type = "prebuilt"
+repo = "dendrite"
+commit = "9963f4bbfcc4cc2a9c4b3e59c392a8f753d458d8"
+sha256 = "c78ffa466fd0b6db6775b774639f0e787f214f6a0ba1e818778779cc0c5ea5a5"
+
+# TODO(nils): This commit is actually wrong - we should be using a different
```

Comment:
Lemme know if this doesn't work - if we need to specify additional info in the buildomat URL, for example.

The current download link is:

```
https://buildomat.eng.oxide.computer/public/file/oxidecomputer/<REPO>/image/<COMMIT>/<SERVICE NAME>
```

Where the service name would just be `dendrite`:

https://github.com/oxidecomputer/dendrite/blob/4f3f9c6617d21164117664650b6fb843bbc28b39/tools/omicron-manifest.toml#L2

Unfortunately, that's not really sufficient to distinguish between the flavors of the same service.

It seems likely that we'll need a way to distinguish these based on "package name" (e.g., `dendrite_tofino_stub`, `dendrite_softnpu`) when picking what packages to download, while still installing packages with the same service name (e.g. `dendrite`)

We can do this pretty easily in the packaging code, by changing

https://github.com/oxidecomputer/omicron/blob/9074a7882503f1c70c23001e22f5369d25725740/package/src/bin/omicron-package.rs#L181-L186

Into:

```rust
                let url = format!(
                    "https://buildomat.eng.oxide.computer/public/file/oxidecomputer/{}/image/{}/{}",
                    repo,
                    commit,
                    package_name,
                );
```

But this will require updating maghemite and dendrite to publish their output files to these URLs based on the package name, not the service name.

Reply:
Aligning maghemite to this format: https://github.com/oxidecomputer/maghemite/pull/20

Reply:
Dendrite appears to [already be doing the right thing](https://github.com/oxidecomputer/dendrite/blob/e5ae7508778d02e079d9a10942ac14ec2064ccb7/.github/buildomat/jobs/image.sh#L14-L32), as long as the packages used in omicron are called `dendrite-stub` and `dendrite-asic`, respectively.

Updating to use these package names.

Reply:
Yes - we've been digging a tunnel from opposite ends.

Reply:
This is done now - we should be using the names emitted by the `dendrite` repo correctly now.

As a heads up, the flow of packages usually goes from:
- BUILD: (optional) Outside this repo, for external packages.
- BUILD: Inside this repo, in the `out` directory
- INSTALL: On the target machine, installed to the `/opt/oxide` directory

I've updated the PR such that during all the "build" phases, the tarballs are named after *package* names, not *service* names. This gives us the ability to distinguish, for example, flavors of Dendrite, so we'd have a file like `out/dendrite-stub.tar.gz` here.

However, in the "install" phases, we're using service names. This means that during installation, we move the file to `/opt/oxide/dendrite.tar.gz`, and refer to it exclusively as "dendrite" from that point onwards.